data.base.api/src/main/java/ru/tinkoff/qa/neptune/data/base/api/data/operations/DataOperation.java

```diff
@@ -0,0 +1,345 @@
+     */
+    public static <T extends PersistableObject> DataOperation<T> deleted(SelectASingle<T, ?> howToSelect) {
+        checkArgument(nonNull(howToSelect), "Please define how to select an object to be deleted");
+        return new DataOperation<T>(format("Deleted %s", howToSelect),
+                DataOperation::delete)
+                .from(context -> {
+                    var result = context.select(howToSelect);
+                    var list = ofNullable(result).map(List::of).orElseGet(List::of);
+                    return getMap(context, list);
+                });
+    }
+
+    /**
+     * Deleting a list of stored records. Records to be deleted are selected by query and then deleted.
+     *
+     * @param howToSelect is a description of query how to select records
+     * @param <T> is a type of {@link PersistableObject} to be deleted
+     * @return an instance of {@link DataOperation}
+     */
+    public static <T extends PersistableObject> DataOperation<T> deleted(SelectList<T, ?> howToSelect) {
+        checkArgument(nonNull(howToSelect), "Please define how to select objects to be deleted");
+        return new DataOperation<T>(format("Deleted %s", howToSelect),
+                DataOperation::delete)
+                .from(context -> getMap(context, context.select(howToSelect)));
+    }
+
+    /**
+     * Deleting a list of stored records.
+     *
+     * @param toBeDeleted is a list of stored records that is selected firstly
+     * @param <T> is a type of {@link PersistableObject} to be deleted
+     * @return an instance of {@link DataOperation}
+     */
+    public static <T extends PersistableObject> DataOperation<T> deleted(Collection<T> toBeDeleted) {
+        checkArgument(nonNull(toBeDeleted),
+                "Collection of objects to be deleted should be defined as a value that differs from null");
+
+        var toDelete = toBeDeleted
+                .stream()
+                .filter(Objects::nonNull)
+                .collect(toList());
+
+        return new DataOperation<T>(format("Deleted %s object/objects from table/tables %s",
+                toDelete.size(),
+                toDelete.stream()
+                        .map(PersistableObject::fromTable)
+                        .distinct()
+                        .collect(toList())),
+                DataOperation::delete)
+                .from(context -> getMap(context, toDelete));
+    }
+
+    /**
+     * Inserting a list of stored records.
+     *
+     * @param toBeInserted is a list of records to be inserted
+     * @param <T> is a type of {@link PersistableObject} to be inserted
+     * @return an instance of {@link DataOperation}
+     */
+    public static <T extends PersistableObject> DataOperation<T> inserted(Collection<T> toBeInserted) {
+        checkArgument(nonNull(toBeInserted),
+                "Collection of objects to be inserted should be defined as a value that differs from null");
+
+        var toInsert = toBeInserted
```

Comment:
Should it be checked like this:

checkArgument(toInsert.size() > 0,
                "At least one object to be inserted should be defined");